### PR TITLE
Update docs zh cn

### PR DESCRIPTION
**Pull request by scloudyy**
This pr update some document in Chinese to new state.

### CODE REVIEW

**Comment by facebook-github-bot-1** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

File: docs/docs/03-interactivity-and-dynamic-uis.zh-CN.md [hunk 1]
```diff
@@ -11,22 +11,26 @@ next: multiple-components-zh-CN.html
...
     return (
-      <p onClick={this.handleClick}>
+      <div onClick={this.handleClick}>
         You {text} this. Click to toggle.
-      </p>
+      </div>
     );
   }
-});
+}
```
Comment (zpao) [inline docs/docs/03-interactivity-and-dynamic-uis.zh-CN.md]: Can you leave this example as is? We're not using ES6 classes broadly yet and this should match the code in the English version.

**Comment by zpao** [comment]
@keyanzhang any chance you could give this a cursory review?

**Comment by facebook-github-bot-5** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

File: docs/docs/03-interactivity-and-dynamic-uis.zh-CN.md [hunk 2]
```diff
@@ -42,9 +46,9 @@ React 里只需把事件处理器（event handler）以骆峰命名（camelCased
 
 在幕后，React 做了一些操作来让代码高效运行且易于理解。
 
-**Autobinding:** 在 JavaScript 里创建回调的时候，为了保证 `this` 的正确性，一般都需要显式地绑定方法到它的实例上。在 React，所有方法被自动绑定到了它的组件实例上（除非使用ES6的class符号）。React 还缓存这些绑定方法，所以 CPU 和内存都是非常高效。而且还能减少打字！
+**自动绑定:** 在 JavaScript 里创建回调的时候，为了保证 `this` 的正确性，一般都需要显式地绑定方法到它的实例上。在 React 中，所有方法被自动绑定到了它的组件实例上（[除非使用ES6的class符号](/react/docs/reusable-components.html#no-autobinding)）。React 还缓存这些绑定方法，所以 CPU 和内存都是非常高效。而且还能减少打字！
 
-**事件代理 ：** React 实际并没有把事件处理器绑定到节点本身。当 React 启动的时候，它在最外层使用唯一一个事件监听器处理所有事件。当组件被加载和卸载时，只是在内部映射里添加或删除事件处理器。当事件触发，React 根据映射来决定如何分发。当映射里处理器时，会当作空操作处理。参考 [David Walsh 很棒的文章](http://davidwalsh.name/event-delegate) 了解这样做高效的原因。
+**事件代理:** React 实际并没有把事件处理器绑定到节点本身。当 React 启动的时候，它在最外层使用唯一一个事件监听器处理所有事件。当组件被加载和卸载时，只是在内部映射里添加或删除事件处理器。当事件触发，React 根据映射来决定如何分发。当映射里没有处理器时，会当作空操作处理。参考 [David Walsh 很棒的文章](http://davidwalsh.name/event-delegate) 了解这样做高效的原因。
```
Comment (keyanzhang) [inline docs/docs/03-interactivity-and-dynamic-uis.zh-CN.md]: 译为「事件处理函数」会比「处理器」更好一些

File: docs/docs/04-multiple-components.zh-CN.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ next: reusable-components-zh-CN.html
 
 目前为止，我们已经学了如何用单个组件来展示数据和处理用户输入。下一步让我们来体验 React 最激动人心的特性之一：可组合性（composability）。
 
-## 动机：关注分离
+## 动机：分离关注
```
Comment (keyanzhang) [inline docs/docs/04-multiple-components.zh-CN.md]: 能否请您将其改回「关注分离」？https://zh.wikipedia.org/wiki/%E5%85%B3%E6%B3%A8%E7%82%B9%E5%88%86%E7%A6%BB

File: docs/docs/04-multiple-components.zh-CN.md [hunk 4]
```diff
@@ -100,7 +100,7 @@ ReactDOM.render(
 </Card>
 // 第二次渲染
 <Card>
-  <p style={{'{{'}}display: 'none'}}>Paragraph 1</p>
+  <p style={{display: 'none'}}>Paragraph 1</p>
```
Comment (keyanzhang) [inline docs/docs/04-multiple-components.zh-CN.md]: 此处之前的写法并没有问题；jekyll 需要我们用 `'{{'}}` 这样的写法。

File: docs/docs/05-reusable-components.zh-CN.md [hunk 5]
```diff
@@ -224,11 +224,38 @@ Counter.defaultProps = { initialCount: 0 };
 
 ### 无自动绑定
 
-方法遵循正式的ES6 class的语义，意味着它们不会自动绑定`this`到实例上。你必须显示的使用`.bind(this)` or [箭头函数](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Functions/Arrow_functions) `=>`.
+方法遵循正式的ES6 class的语义，意味着它们不会自动绑定`this`到实例上。你必须显示的使用`.bind(this)` or [箭头函数](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Functions/Arrow_functions) `=>`：
+
+```javascript
+// 你可以使用 bind() 来保护 `this`
```
Comment (keyanzhang) [inline docs/docs/05-reusable-components.zh-CN.md]: 此处译为「绑定」会比「保护」更好一些

File: docs/docs/05-reusable-components.zh-CN.md [hunk 6]
```diff
@@ -224,11 +224,38 @@ Counter.defaultProps = { initialCount: 0 };
...
+```
+
+现在你可以直接使用 `this.tick` 因为它已经在构造函数里绑定过一次了。
+
+```javascript
+// 它已经在构造函数里绑定过了
+<div onClick={this.tick}>
+```
+
+这对应用的性能有帮助，特别是当你用 [深比较](/react/docs/shallow-compare.html) 实现 [shouldComponentUpdate()](/react/docs/component-specs.html#updating-shouldcomponentupdate) 时。
```
Comment (keyanzhang) [inline docs/docs/05-reusable-components.zh-CN.md]: 深比较 -> 浅层比较

File: docs/docs/05-reusable-components.zh-CN.md [hunk 7]
```diff
@@ -249,7 +276,7 @@ ReactDOM.render(<HelloMessage name="Sebastian" />, mountNode);
 ```
 
 这个简化的组件API旨在用于那些纯函数态的组件 。这些组件必须没有保持任何内部状态，没有备份实例，也没有组件生命周期方法。他们纯粹的函数式的转化他们的输入，没有引用。
-然而，你仍然可以以设置为函数的properties的方式来指定 `.propTypes` 和 `.defaultProps`，就像你在ES6类里设置他们那样。
+然而，你仍然可以以设置为函数的 properties 的方式来指定 `.propTypes` 和 `.defaultProps`，就像你在ES6类里设置他们那样。
```
Comment (keyanzhang) [inline docs/docs/05-reusable-components.zh-CN.md]: 「设置函数 properties 的方式」

File: docs/docs/07-forms.zh-CN.md [hunk 8]
```diff
@@ -102,7 +102,7 @@ next: working-with-the-browser-zh-CN.html
   }
 ```
 
-这个例子会像上面的 **不受控组件** 例子一样运行。
+这个例子会像上面的 **受控组件** 例子一样运行。
```
Comment (keyanzhang) [inline docs/docs/07-forms.zh-CN.md]: Thanks for catching this!

File: docs/docs/08-working-with-the-browser.zh-CN.md [hunk 9]
```diff
@@ -8,17 +8,17 @@ next: more-about-refs-zh-CN.html
 
 React提供了强大的抽象机制使你在大多数情况下免于直接接触DOM，但有时你仅仅只需要访问底层API，也许是为了与第三方库或者已有的代码协作。
 
-## 虚拟DOM
+## 虚拟 DOM
 
-React非常快速是因为它从不直接操作DOM。React维持了一个快速的内存中的DOM表示。`render()` 方法实际上返回一个对DOM的*描述*，然后React能根据内存中的“描述”来比较此“描述”以计算出最快速的方法更新浏览器。
+React非常快速是因为它从不直接操作DOM。React维持了一个快速的内存中的DOM表示。`render()` 方法实际上返回一个对DOM的*描述*，然后React能将其与内存中的“描述”进行以计算出最快速的方式更新浏览器。
```
Comment (keyanzhang) [inline docs/docs/08-working-with-the-browser.zh-CN.md]: 「进行比较，以计算出」

**Comment by keyanzhang** [comment]
Thanks for keeping our Chinese docs up-to-date with the English version @scloudyy! This is awesome work. Could you revise the parts we've mentioned in the inline comments? 

Also seems there are some merge conflicts so you'll need to rebase on master and force push. Thanks!

**Comment by scloudyy** [comment]
It seems OK now @keyanzhang

**Comment by keyanzhang** [comment]
Thanks @scloudyy!

**Comment by scloudyy** [comment]
I have changed back to the old code, but it seems that the English version uses ES6. And it is very strange that only in this part the English version uses ES6 @zpao

**Comment by zpao** [comment]
Oh wow, you are 100% correct, I'm sorry - that code is in ES6 now. I didn't think we had made that change. Feel free to put it back in!

**Comment by Hasanul-Hoque2013** [comment]
Hasanul Hoque


Hasanul Hoque

**Comment by keyanzhang** [comment]
Thanks @scloudyy! Again this is super great work. We plan to improve our internationalization for documentation so please help us to further improve it when you have time!

**Pull request merged**